src/hotspot/share/opto/printinlining.hpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef PRINTINLINING_HPP
+#define PRINTINLINING_HPP
+
+#include "memory/allocation.hpp"
+#include "utilities/ostream.hpp"
+#include "utilities/growableArray.hpp"
+#include "nmt/nmtTreap.hpp"
+
+class JVMState;
+class ciMethod;
+class Compile;
+enum class InliningResult;
+
+// If not enabled, all method calls are no-ops.
+class InlinePrinter {
+private:
+  class IPInlineAttempt {
+  private:
+    InliningResult _result;
+    stringStream* _stream = nullptr;
+
+  public:
+    IPInlineAttempt() {}
```

Comment:
Since this is a default constructor, you could probably use:
```suggestion
    IPInlineAttempt() = default;
```